src/components/nodes-table/nodes-table.js

```diff
@@ -0,0 +1,256 @@
+    nodeTableItems: undefined,
+    selectedNode: undefined,
+    filters: {},
+    sorting: {},
+    headers: [
+      { text: 'ID', value: 'id' },
+      { text: 'Manufacturer', value: 'manufacturer' },
+      { text: 'Product', value: 'productDescription' },
+      { text: 'Product code', value: 'productLabel' },
+      { text: 'Name', value: 'name' },
+      { text: 'Location', value: 'loc' },
+      { text: 'Secure', value: 'isSecure' },
+      { text: 'Beaming', value: 'isBeaming' },
+      { text: 'Failed', value: 'failed' },
+      { text: 'Status', value: 'status' },
+      { text: 'Interview stage', value: 'interviewStage' },
+      { text: 'Last Active', value: 'lastActive' }
+    ]
+  }),
+  methods: {
+    initFilters () {
+      return {
+        id: { type: 'number' },
+        manufacturer: { type: 'string' },
+        productDescription: { type: 'string' },
+        productLabel: { type: 'string' },
+        name: { type: 'string' },
+        loc: { type: 'string' },
+        isSecure: { type: 'boolean' },
+        isBeaming: { type: 'boolean' },
+        failed: { type: 'boolean' },
+        status: { type: 'string' },
+        interviewStage: { type: 'string' },
+        lastActive: { type: 'date' }
+      }
+    },
+    initSorting () {
+      return {
+        by: ['node_id'],
+        desc: [false]
+      }
+    },
+    loadSetting (key, defaultVal) {
+      return this.settings.load(key, defaultVal)
+    },
+    storeSetting (key, val) {
+      this.settings.store(key, val)
+    },
+    resetFilter () {
+      this.filters = this.initFilters()
+    },
+    nodeSelected (node) {
+      this.selectedNode = node
+      this.$emit('node-selected', { node })
+    }
+  },
+  mounted () {
+    this.filters = this.loadSetting('nodes_filters', this.initFilters())
+    this.sorting = this.loadSetting('nodes_sorting', this.initSorting())
+    this.nodeTableItems = this.loadSetting('nodes_itemsPerPage', 10)
+  },
+  watch: {
+    nodeTableItems (val) {
+      this.storeSetting('nodes_itemsPerPage', val)
+    },
+    filters: {
+      handler (val) {
+        this.storeSetting('nodes_filters', val)
+      },
+      deep: true
+    },
+    sorting: {
+      handler (val) {
+        this.storeSetting('nodes_sorting', val)
+      },
+      deep: true
+    }
+  },
+  computed: {
+    nodeCollection () {
+      return new NodeCollection(this.nodes)
+    },
+    relevantNodes () {
+      return this.nodeCollection.filter('failed', failed => {
+        return this.showHidden ? true : !failed
+      })
+    },
+    filteredNodes () {
+      return this.relevantNodes
+        .betweenNumber(
+          'id',
+          this.filters.id ? this.filters.id.min : null,
+          this.filters.id ? this.filters.id.max : null
+        )
+        .equalsAny(
+          'id',
+          this.filters.id
+            ? this.filters.id.selections
+              ? this.filters.id.selections
+              : []
+            : []
```

Comment:
@ahochsteger make this cleaner

Reply:
All this pattern are repeated. EqualsAny should receive only `(property, filters)`

Reply:
@robertsLando, this PR is still WIP and not yet ready for review.
I already have a version locally that has all repeated sections completely removed and refactored to be as generic as possible.
There are some issues I have to resolve yet until it's ready to be pushed.

Reply:
@ahochsteger Ok!